src/lazy/encoder/container_fn.rs

```diff
@@ -0,0 +1,56 @@
+//! Container population traits that allow closures to be used in places where the borrow checker
+//! would normally balk due to point-in-time limitations.
+//! See: <https://blog.rust-lang.org/2022/10/28/gats-stabilization.html#implied-static-requirement-from-higher-ranked-trait-bounds>
+
+use crate::lazy::encoder::value_writer::ValueWriter;
+use crate::IonResult;
+
+/// Generates a trait with a single method that takes one of [`ValueWriter`]'s associated types as
+/// an argument. The trait extends `FnOnce(TheAssociatedType) -> IonResult<()>`, allowing it to be
```

Comment:
This is just a suggestion, feel free to take it or leave it! I think it might be worth adding whats the functionality of `populate` for this workaround. 

```suggestion
/// Generates a trait with a single method that takes one of [`ValueWriter`]'s associated types as
/// an argument. This `populate` method has the same functionality as the closure and it helps populate the container by adding new fields/values. The trait extends `FnOnce(TheAssociatedType) -> IonResult<()>`, allowing it to be
```